CMakeLists.txt

```diff
@@ -439,14 +439,15 @@ add_subdirectory(build_tools/third_party/flatcc EXCLUDE_FROM_ALL)
 add_subdirectory(build_tools/third_party/half EXCLUDE_FROM_ALL)
 add_subdirectory(build_tools/third_party/pffft EXCLUDE_FROM_ALL)
 add_subdirectory(build_tools/third_party/renderdoc_api EXCLUDE_FROM_ALL)
-add_subdirectory(build_tools/third_party/ruy EXCLUDE_FROM_ALL)
 add_subdirectory(build_tools/third_party/vulkan_memory_allocator EXCLUDE_FROM_ALL)
 
 add_subdirectory(third_party/cpuinfo EXCLUDE_FROM_ALL)
 add_subdirectory(third_party/googletest EXCLUDE_FROM_ALL)
 add_subdirectory(third_party/abseil-cpp EXCLUDE_FROM_ALL)
 add_subdirectory(third_party/flatcc EXCLUDE_FROM_ALL)
 add_subdirectory(third_party/vulkan_headers EXCLUDE_FROM_ALL)
+# ruy must come after cpuinfo and googletest so it uses already-defined targets.
+add_subdirectory(third_party/ruy EXCLUDE_FROM_ALL)
```

Comment:
If you guarded cpuinfo with the existence of its CMakeLists.txt as well, this wouldn't be required.

Reply:
Cpuinfo is currently a mandatory dependency of Ruy (as far as the CMakeLists is concerned --- Bazel is different due to supporting some exotic platforms, but ruy's performance is degraded when cpuinfo is not available, and since Ruys CMake users care about performance, I didn't see a need to support cpuinfo-less builds in ruy cmakelists).

Sounds ok?

Reply:
I'm not saying you should support cpuinfo-less builds of ruy, but you can support someone providing their own cpuinfo (as IREE does). If the ruy cpuinfo submodule isn't checked out, then that is a strong indicator that someone doesn't want to use the cpuinfo that comes from ruy's submodule (and must provide their own). Assuming the build just hard fails with "I can't find cpuinfo" (as opposed to silently degrading performance), then not adding the directory when it doesn't exist is nicer than requiring that ruy be added in some particular order.